omod/src/main/java/org/openmrs/module/reportingrest/web/resource/ReportRequestResource.java

```diff
@@ -65,21 +72,80 @@ public ReportRequest newDelegate() {
 	public ReportRequest getByUniqueId(String uuid) {
 		return getService().getReportRequestByUuid(uuid);
 	}
-	
+
 	@Override
 	protected PageableResult doSearch(RequestContext context) {
+		String statusesGroup = context.getParameter("statusesGroup");
+		if (StringUtils.isNotBlank(statusesGroup)) {
+			List<ReportRequest.Status> statuses = new ArrayList<ReportRequest.Status>();
+			if (StringUtils.equalsIgnoreCase(statusesGroup, "ran")) {
+				Collections.addAll(
+						statuses,
+						ReportRequest.Status.REQUESTED,
+						ReportRequest.Status.COMPLETED,
+						ReportRequest.Status.SAVED,
+						ReportRequest.Status.FAILED,
+						ReportRequest.Status.PROCESSING);
+			} else if (StringUtils.equalsIgnoreCase(statusesGroup, "processing")) {
+				Collections.addAll(statuses, ReportRequest.Status.REQUESTED, ReportRequest.Status.PROCESSING);
+			} else if (StringUtils.equalsIgnoreCase(statusesGroup, "scheduled")) {
+				Collections.addAll(
+						statuses,
+						ReportRequest.Status.SCHEDULED,
+						ReportRequest.Status.SCHEDULE_COMPLETED);
+			}
+			ReportService reportService = getService();
+			Integer pageNumber = context.getStartIndex();
+			Integer pageSize = context.getLimit();
+			ReportRequestDTO reportRequestsDTO = reportService.getReportsWithPagination(
+					null, null, null, pageNumber, pageSize, statuses.toArray(new ReportRequest.Status[0]));
+			List<ReportRequest> reportRequests = reportRequestsDTO.getReportRequests();
+
+			String sortByParameter = context.getParameter("sortBy");
+			if (StringUtils.isBlank(sortByParameter) || "priority".equals(sortByParameter)) {
+				Collections.sort(reportRequests, new ReportRequest.PriorityComparator());
+				Collections.reverse(reportRequests);
+			} else if("name".equals(sortByParameter)) {
+				Collections.sort(reportRequests, new Comparator<ReportRequest>() {
+					@Override
+					public int compare(ReportRequest left, ReportRequest right) {
+						return left
+								.getReportDefinition()
+								.getParameterizable()
+								.getName()
+								.compareTo(right.getReportDefinition().getParameterizable().getName());
+					}
+				});
+			}
+
+			for (ReportRequest reportRequest : reportRequests) {
+				for (RenderingMode mode :
+						reportService.getRenderingModes(
+								reportRequest.getReportDefinition().getParameterizable())) {
+					if (OpenmrsUtil.nullSafeEquals(mode, reportRequest.getRenderingMode())) {
+						reportRequest.setRenderingMode(mode);
+					}
+				}
+			}
+
+			return new AlreadyPaged<ReportRequest>(context, reportRequests, reportRequestsDTO.getReportRequestCount() > (long) pageNumber * pageSize, reportRequestsDTO.getReportRequestCount());
+		}

```

Comment:
I don't understand the placement of this if/then block.  Let's just do things consistently and always use the paging endpoint.  If no paging parameters are supplied, it should default to returning all like it currently does.